api/user/userRouter.js

```diff
@@ -5,8 +5,8 @@ const User = require('./userModel');
 const router = express.Router();
 
 router.get('/', authRequired, function (req, res) {
-  const { role, profile_id } = req.profile;
-  User.findUserData(role, profile_id)
+  const { role_id, profile_id } = req.profile;
+  User.findUserData(role_id, profile_id)
     .then((user) => {
       res.status(200).json(user);
```

Comment:
Instead of writing the helper function within userModel you could also change this line to 'res.status(200).json(user[0]);' if you wanted to extract the object out of the array for the response. This way it'll save a whole extra function from the userModel, unless you wanted to explicitly check for the type Array in which case your way works fine. :)

Reply:
Hi @aaronpear, thanks for your feedback. I originally wrote the solution as you described it, but ultimately felt the above solution would make the code base clearer to understand given that `findUserData` seems to imply that only a single user is returned.

Reply:
Just weighing in here.  I think the "findUserData" is perhaps way more readable and cleaner so we could leave it. But I think reading "user[0]" is also pretty straightforward implying just one result from the array. So I would think saving a function for just a little less clarity here is worth it.

Reply:
@luisbryandr To clarify, prior to the changes introduced in this PR, `findUserData` always returned an array of just one user. Since there was only ever one user in the array, it seemed superfluous to return an array at all user when the user himself could be returned instead. This also has the added benefit of being more DRY, since extracting the user from the array using `user[0]` would only need to be written in one place (inside the `findUserData` function).